app/services/editions/publishing/PublishedIssuesBucket.scala

```diff
@@ -2,31 +2,14 @@ package services.editions.publishing
 
 import java.time.OffsetDateTime
 
-import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import com.amazonaws.services.s3.AmazonS3
 import com.gu.editions.PublishedIssue
-import conf.ApplicationConfiguration
-import play.api.libs.json.Json
-import services._
-import PublishedIssueFormatters._
 import com.gu.pandomainauth.model.User
 
-class PublishedIssuesBucket(config: ApplicationConfiguration, awsEndpoints: AwsEndpoints) {
-
-  val bucket = config.aws.publishedEditionsIssuesBucket
-
-  val client = AmazonS3ClientBuilder
-    .standard()
-    .withCredentials(config.aws.credentials.get)
-    .withRegion(config.aws.region)
-    .build()
-
-  private def createIssuePath(issue: PublishedIssue, user: User) = {
-    val firstName = user.firstName.replace(" ", "_")
-    val lastName = user.lastName.replace(" ", "_")
-    s"${issue.name}/${issue.issueDate.toLocalDate.toString}/${OffsetDateTime.now().toString}_${firstName}_$lastName.json"
-  }
-
-  def putIssue(issue: PublishedIssue, user: User) = {
-    client.putObject(bucket, createIssuePath(issue, user), Json.stringify(Json.toJson(issue)))
+class PublishedIssuesBucket(s3Client: AmazonS3, bucketName: String) extends EditionsBucket[User](s3Client, bucketName) {
+  override def createIssueFilename(issue: PublishedIssue, publicationMetadata: User): String = {
+    val firstName = publicationMetadata.firstName.replace(" ", "_")
+    val lastName = publicationMetadata.lastName.replace(" ", "_")
+    s"${OffsetDateTime.now().toString}_${firstName}_$lastName.json"
```

Comment:
do we need the `firstName` and `lastName` on the filename?

Reply:
Killed - need to check with @AWare prior to merging but fairly sure he doesn't rely on the user in the file name.

Reply:
@AWare is on holiday till Thurs so I'm going to merge and seek forgiveness later...